eng/scripts/automation/Invoke-GenerateAndBuild.ps1

```diff
@@ -69,53 +68,50 @@ if ($readmeFile) {
+  foreach ($typespecRelativeFolder in $relatedTypeSpecProjectFolder) {
+      $typespecFolder = Resolve-Path (Join-Path $swaggerDir $typespecRelativeFolder)
+      $processScript = Resolve-Path (Join-Path $sdkPath "eng/common/scripts" "TypeSpec-Project-Process.ps1")
+      $sdkProjectFolders = Get-ChildItem -Path (Join-Path $sdkPath "sdk") -Depth 1 -Directory | Select-Object -ExpandProperty FullName
 
-  $tspConfigYaml = Get-Content -Path (Join-Path "$typespecFolder" "tspconfig.yaml") -Raw
-
-  Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
-  $yml = ConvertFrom-YAML $tspConfigYaml
-  $service = ""
-  $namespace = ""
-  if ($yml) {
-      if ($yml["parameters"] -And $yml["parameters"]["service-directory-name"]) {
-          $service = $yml["parameters"]["service-directory-name"]["default"];
-      }
-      if ($yml["options"] -And $yml["options"]["@azure-tools/typespec-csharp"] -And $yml["options"]["@azure-tools/typespec-csharp"]["namespace"]) {
-          $namespace = $yml["options"]["@azure-tools/typespec-csharp"]["namespace"]
+      # Invoke Process script. SkipSyncAndGenerate only when it's not a new SDK project
+      $sdkProjectFolder = & $processScript $typespecFolder $commitid $repoHttpsUrl -SkipSyncAndGenerate
+      if ($LASTEXITCODE) {
+        # If Process script call fails, then return with failure to CI and don't need to call GeneratePackage
+        $generatedSDKPackages.Add(@{
+          result = "failed";
+          path=@("");
+        })
+      } else {
+          $relativeSdkPath = Resolve-Path $sdkProjectFolder -Relative
+          $specRepoRoot = $swaggerDir -replace "/specification", ""
+          if ($sdkProjectFolders -contains $sdkProjectFolder) {
+            # Existed SDK project case, needs to generate code
+            GeneratePackage `
+            -projectFolder $sdkProjectFolder `
+            -sdkRootPath $sdkPath `
+            -path $relativeSdkPath `
+            -downloadUrlPrefix $downloadUrlPrefix `
+            -serviceType "data-plane" `
+            -generatedSDKPackages $generatedSDKPackages `
+            -specRepoRoot $specRepoRoot
+          } else {
+            # New SDK project case, code is already generated by emitter. So, skip code generation
+            GeneratePackage `
+            -projectFolder $sdkProjectFolder `
+            -sdkRootPath $sdkPath `
+            -path $relativeSdkPath `
+            -downloadUrlPrefix $downloadUrlPrefix `
+            -serviceType "data-plane" `
+            -skipGenerate `
+            -generatedSDKPackages $generatedSDKPackages `
+            -specRepoRoot $specRepoRoot
+          }
       }
   }
-  if (!$service || !$namespace) {
-      throw "Not provide service name or namespace."
-  }
-  $projectFolder = (Join-Path $sdkPath "sdk" $service $namespace)
-  $specRoot = $swaggerDir
-  if ((-Not $relatedTypeSpecProjectFolder.Contains("specification")) -And $swaggerDir.Contains("specification"))
```

Comment:
Invoke-GenerateAndBuild.ps1 is used to handle both azure-rest-api-spec  repo which contains `specification` folder and service dev oneBranch repo which does not contains `specification` folder. That's why we need this logic to support two environment.
Make sure you remove this will not break anything.

Reply:
As we don't have real service team using this in their ADO repo and shift left project is suspended, I would rather ignore this scenario for now.